server/src/main/java/org/elasticsearch/action/search/ExpandSearchPhase.java

```diff
@@ -60,6 +72,9 @@ public void run() {
             MultiSearchRequest multiRequest = new MultiSearchRequest();
             if (collapseBuilder.getMaxConcurrentGroupRequests() > 0) {
                 multiRequest.maxConcurrentSearchRequests(collapseBuilder.getMaxConcurrentGroupRequests());
+            } else if (context.getRequest().preference() != null) {
```

Comment:
Do we need to preserve the preference parameter for the expanded search requests? Can we discard it and route the expanded requests to all replicas?